homeassistant/util/json.py

```diff
@@ -60,29 +61,14 @@ def save_json(
         _LOGGER.error(msg)
         raise SerializationError(msg) from error
 
-    tmp_filename = ""
-    tmp_path = os.path.split(filename)[0]
     try:
-        # Modern versions of Python tempfile create this file with mode 0o600
-        with tempfile.NamedTemporaryFile(
-            mode="w", encoding="utf-8", dir=tmp_path, delete=False
-        ) as fdesc:
-            fdesc.write(json_data)
-            tmp_filename = fdesc.name
-        if not private:
-            os.chmod(tmp_filename, 0o644)
-        os.replace(tmp_filename, filename)
+        with atomicwrites.atomic_write(filename, overwrite=True) as fp:
+            fp.write(json_data)
+
+        os.chmod(filename, 0o600 if private else 0o644)
```

Comment:
```suggestion
            os.chmod(fp, 0o600 if private else 0o644)
```

Reply:
I don't think there is a reason to put it inside the atomic writes context because it doesn't use anything from the context manager.

Reply:
It makes sure that we have the right user access rights even if a power failure happens. In theory, otherwise, we could end up with the new file but still with the wrong access mods.

Reply:
@balloob it uses `fp`. And with that, it is executed before the rename and fsync on the parent folder.